js/__internal/grids/grid_core/module_types.ts

```diff
@@ -1,13 +1,28 @@
 /* eslint-disable max-classes-per-file */
-import { PropertyType as _PropertyType, DeepPartial, OmitInternal } from '@js/core/index';
+import {
+  PropertyType as _PropertyType,
+  DeepPartial,
+  OmitInternal,
+} from '@js/core/index';
 import { Component } from '@js/core/component';
 import { dxElementWrapper } from '@js/core/renderer';
 import DataGrid, { Properties } from '@js/ui/data_grid';
 
-type PropertyType<O, K extends string> = _PropertyType<O, K> extends never ? any : _PropertyType<O, K>;
-
-export interface InternalGrid extends Omit<DataGrid<unknown, unknown>, 'option'> {
-  option: (<TPropertyName extends string>(optionName: TPropertyName) => PropertyType<InternalGridOptions, TPropertyName>) & (<TPropertyName extends string>(optionName: TPropertyName, optionValue: PropertyType<InternalGridOptions, TPropertyName>) => void) & (() => InternalGridOptions) & ((options: InternalGridOptions) => void);
+type PropertyType<O, K extends string> = _PropertyType<O, K> extends never
+  ? any
+  : _PropertyType<O, K>;
+
+export interface InternalGrid
+  extends Omit<DataGrid<unknown, unknown>, 'option'> {
+  option: (<TPropertyName extends string>(
+    optionName: TPropertyName
+  ) => PropertyType<InternalGridOptions, TPropertyName>) &
+  (<TPropertyName extends string>(
+    optionName: TPropertyName,
+    optionValue: PropertyType<InternalGridOptions, TPropertyName>
+  ) => void) &
+  (() => InternalGridOptions) &
+  ((options: InternalGridOptions) => void);
```

Comment:
Let's describe each type as a separate type to make it more obvious

E.g
```typescript
type GetOptionValueType = <TPropertyName extends string>(
  optionName: TPropertyName
) => PropertyType<InternalGridOptions, TPropertyName>;
type SetOptionValueType = <TPropertyName extends string>(
  optionName: TPropertyName,
  optionValue: PropertyType<InternalGridOptions, TPropertyName>
) => void;
type GetOptionsType = () => InternalGridOptions;
type SetOptionsType = (options: InternalGridOptions) => void;
```

and then
```suggestion
  option: GetOptionValueType 
  & SetOptionValueType 
  & GetOptionsType 
  & SetOptionsType;
```